discovery/pkg/openstack/reconciler.go

```diff
@@ -117,48 +123,64 @@ func (r *Reconciler) reconcile() {
 		// Get load balancers that are defined in the project
 		loadbalancers, err := r.ListLoadBalancers(project.ID)
 		if err != nil {
-			r.Metrics.GenericMetricError(r.BackendName, "ListLoadBalancers")
+			r.Metrics.GenericMetricError("ListLoadBalancers")
 			log.Errorf("error reconciling project %q: %v", projectName, err)
 			continue
 		}
 
+		totalUpstreamServices := len(loadbalancers)
 		loadbalancers = r.skipInvalidLoadBalancers(projectName, loadbalancers)
+		totalInvalidServices := totalUpstreamServices - len(loadbalancers)
 
 		// Get all pools defined in the project
 		pools, err := r.ListPools(project.ID)
 		if err != nil {
-			r.Metrics.GenericMetricError(r.BackendName, "ListPools")
+			r.Metrics.GenericMetricError("ListPools")
 			log.Errorf("error reconciling project %q: %v", projectName, err)
 			continue
 		}
 
-		// Get all services and endpoints that exist in the corresponding
-		// namespace
+		// Get all services and endpoints that exist in the corresponding namespace
 		clusterLabelSelector := fmt.Sprintf("%s=%s", translator.GimbalLabelBackend, r.BackendName)
 		currentServices, err := r.GimbalKubeClient.CoreV1().Services(projectName).List(metav1.ListOptions{LabelSelector: clusterLabelSelector})
 		if err != nil {
-			r.Metrics.GenericMetricError(r.BackendName, "ListServicesInNamespace")
+			r.Metrics.GenericMetricError("ListServicesInNamespace")
 			log.Errorf("error listing services in namespace %q: %v", projectName, err)
 			continue
 		}
 
-		currentEndpoints, err := r.GimbalKubeClient.CoreV1().Endpoints(projectName).List(metav1.ListOptions{LabelSelector: clusterLabelSelector})
+		currentk8sEndpoints, err := r.GimbalKubeClient.CoreV1().Endpoints(projectName).List(metav1.ListOptions{LabelSelector: clusterLabelSelector})
 		if err != nil {
-			r.Metrics.GenericMetricError(r.BackendName, "ListEndpointsInNamespace")
+			r.Metrics.GenericMetricError("ListEndpointsInNamespace")
 			log.Errorf("error listing endpoints in namespace:%q: %v", projectName, err)
 			continue
 		}
 
+		// Convert the k8s list to type []Endpoints so make comparison easier
+		currentEndpoints := []Endpoints{}
+		for _, v := range currentk8sEndpoints.Items {
+			currentEndpoints = append(currentEndpoints, Endpoints{endpoints: v, upstreamName: ""})
```

Comment:
Do we need to set the `upstreamName` here?

Reply:
Those are the current endpoints which come from the k8s query. The upstream name is not used, but having two lists of the same type makes the comparison logic work as-is.